src/apps/copilots/src/services/copilot-opportunities.ts

```diff
@@ -85,3 +90,38 @@ export const useCopilotOpportunity = (opportunityId?: string): CopilotOpportunit
         revalidateOnFocus: false,
     })
 }
+
+/**
+ * apply copilot opportunity
+ * @param opportunityId
+ * @param request
+ * @returns
+ */
+export const applyCopilotOpportunity = async (opportunityId: number, request: {
+    notes?: string;
+}): Promise<CopilotApplication> => {
+    const url = `${copilotBaseUrl}/copilots/opportunity/${opportunityId}/apply`
+
+    return xhrPostAsync(url, request, {})
+}
+
+/**
+ * Custom hook to fetch copilot applications by opportunity id.
+ *
+ * @param {string} opportunityId - The unique identifier of the copilot request.
+ * @returns {CopilotApplicationResponse} - The response containing the copilot application data.
+ */
+export const useCopilotApplications = (opportunityId?: string): CopilotApplicationResponse => {
+    const url = opportunityId
+        ? buildUrl(`${copilotBaseUrl}/copilots/opportunity/${opportunityId}/applications`)
+        : undefined
+
+    const fetcher = (urlp: string): Promise<CopilotApplication[]> => xhrGetAsync<CopilotApplication[]>(urlp)
+        .then(data => data)
+        .catch(() => [])
+
+    return useSWR(url, fetcher, {
```

Comment:
Please remove allow SWR to automatically revalidates data - no need to block that here.